src/stats/stats.ts

```diff
@@ -0,0 +1,85 @@
+import _ from "lodash";
+
+import { FrameEntryType, PlayerIndexedType, Frames, FramesType } from "./common";
+
+export interface StatComputer<T> {
+    setPlayerIndices(indices: PlayerIndexedType[]): void;
+    processFrame(newFrame: FrameEntryType, allFrames: FramesType): void;
+    fetch(): T;
+}
+
+export interface StatOptions {
+    processOnTheFly: boolean;
+}
+
+const defaultOptions: StatOptions = {
+    processOnTheFly: false,
+}
+
+export class Stats {
+    private options: StatOptions;
+    private lastProcessedFrame: number | null = null;
+    private lastFrame: number;
+    private frames: FramesType = {};
+    private opponentIndices = new Array<PlayerIndexedType>();
+    private allComputers = new Array<StatComputer<unknown>>();
+
+    public constructor(options?: StatOptions) {
+        this.options = options || defaultOptions;
+    }
+
+    public setPlayerIndices(indices: PlayerIndexedType[]): void {
+        this.opponentIndices = indices;
+        this.allComputers.forEach(comp => comp.setPlayerIndices(indices));
+    }
+
+    public register(computer: StatComputer<unknown>): void {
+        this.allComputers.push(computer);
+    }
+
+    public registerAll(computers: StatComputer<unknown>[]): void {
+        this.allComputers = this.allComputers.concat(computers);
+    }
+
+    public process(): void {
+        // Finish processing if we're not up to date
+        if (this.lastFrame === this.lastProcessedFrame) {
+            return;
+        }
+        if (this.opponentIndices.length === 0) {
+            return;
+        }
+        let i = this.lastProcessedFrame ? this.lastProcessedFrame + 1 : Frames.FIRST;
+        while (Boolean(this.frames[i])) {
```

Comment:
Probably not necessary to use `Boolean()` here, when you put something like this in a while loop it's already implied that it's being interpreted as a boolean. Not blocking or important, but just seems a bit too explicit

Reply:
Removed the `Boolean` assertion in commit 5016f1.